tests/run_tests.sh

```diff
@@ -17,17 +9,59 @@ if [ -n "$1" ]; then
 	shift
 fi
 
+if [ "--help" = "$MARK" -o "-h" = "$MARK" ]; then
+	cat <<EOHELP
+
+USAGE
+
+   run_tests.sh -h|--help
+   run_tests.sh <selection>
+
+DESCRIPTION
+
+  Simple command to run tests using installed bcbio_nextgen python/nose. Pass optional argument to select type of tests to run:
+
+    ./run_tests.sh rnaseq
+    ./run_tests.sh speed=1
+    ./run_tests.sh devel
+    ./run_tests.sh docker
+    ./run_tests.sh docker_ipython
+
+ENVIRONMENT
+
+  The script unsets values for PYTHONHOME and PYTHONPATH.
+
+  Use PYTEST to specify an alternative path to the py.test executable.
+
+EOHELP
+	exit 1
+fi
+
 if [[ ${MARK} == docker* && "`which bcbio_vm.py`" != "" ]]; then
-    BCBIO_DIR=$(dirname "$(readlinkf `which bcbio_vm.py`)")
+	BCBIO_DIR=$(dirname "$(readlinkf `which bcbio_vm.py`)")
 else
-    BCBIO_DIR=$(dirname "$(readlinkf `which bcbio_nextgen.py`)")
+	BCBIO_DIR=$(dirname "$(readlinkf `which bcbio_nextgen.py`)")
 fi
 
 unset PYTHONHOME
 unset PYTHONPATH
 export PYTHONNOUSERSITE=1
+
 # Ensure version.py exists in raw cloned bcbio directory
 if [ -d "../bcbio/pipeline" ]; then
 	[ -f ../bcbio/pipeline/version.py ] || touch ../bcbio/pipeline/version.py
 fi
-"$BCBIO_DIR/py.test" -p no:cacheprovider -p no:stepwise -v -s -m ${MARK} "$@"
+
+if [ -z "$PYTEST" ]; then
+	for p in "$BCBIO_DIR/py.test" /usr/bin/py.test-3 /usr/bin/py.test
```

Comment:
Thanks! It is probably better to just check the `py.test` in bcbio and then bail out if it isn't there; falling back to system installed pythons has been a source of pain in the past so we try to keep everything in bcbio completely independent from whatever is installed on the system.

Reply:
I think to have simplified it accordingly. The PYPATH environment is still allowed to contribute, was not sure if you mind that. You know about the background of this patch to have it all compatible for Debian - and for it there is no difference between bcbio's python and the system-installed one - except for the path.